src/instance.ts

```diff
@@ -26,9 +28,32 @@ export class ExtensionInstance {
 	constructor(context: ExtensionContext) {
 		this.context = context;
 
-		this.networkComponent = new NetworkComponent(this);
+		const controlInstance = spawn("node", [ 
```

Comment:
create a new class method for instantiating the child process, this is a messy implementation that we can clean up by making additional functions;

```ts
instantiateExtensionProcess() { ... }
```

Reply:
Will do! This should rather be a function instead of a class, it is intended to not have a prototype.

Reply:
I've moved this to a "module" function (single-use functions) in https://github.com/4x8Matrix/OutSi-TS-Plugin/pull/12/commits/012597baa994ee147b7a2005d0607c369066c822. Hope this improves the readability.